src/Microsoft.Extensions.DependencyModel/DependencyContextLoader.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.Extensions.DependencyModel
+{
+    public class DependencyContextLoader
+    {
+        private const string DepsJsonExtension = ".deps.json";
+        private const string DepsExtension = ".deps";
+
+        private readonly string _entryPointDepsLocation;
+        private readonly string _runtimeDepsLocation;
+        private readonly IFileSystem _fileSystem;
+        private readonly IDependencyContextReader _jsonReader;
+        private readonly IDependencyContextReader _csvReader;
+
+        public DependencyContextLoader() : this(
+            GetDefaultEntrypointDepsLocation(),
+            GetDefaultRuntimeDepsLocation(),
+            FileSystemWrapper.Default,
+            new DependencyContextJsonReader(),
+            new DependencyContextCsvReader())
+        {
+        }
+
+        internal DependencyContextLoader(
+            string entryPointDepsLocation,
+            string runtimeDepsLocation,
+            IFileSystem fileSystem,
+            IDependencyContextReader jsonReader,
+            IDependencyContextReader csvReader)
+        {
+            _entryPointDepsLocation = entryPointDepsLocation;
+            _runtimeDepsLocation = runtimeDepsLocation;
+            _fileSystem = fileSystem;
+            _jsonReader = jsonReader;
+            _csvReader = csvReader;
+        }
+
+        public static DependencyContextLoader Default { get; } = new DependencyContextLoader();
+
+        internal virtual bool IsEntryAssembly(Assembly assembly)
+        {
+            return assembly.GetName() == Assembly.GetEntryAssembly().GetName();
+        }
+
+        internal virtual Stream GetResourceStream(Assembly assembly, string name)
+        {
+            return assembly.GetManifestResourceStream(name);
+        }
+
+        public DependencyContext Load(Assembly assembly)
+        {
+            if (assembly == null)
+            {
+                throw new ArgumentNullException(nameof(assembly));
+            }
+
+            DependencyContext context = null;
+
+            if (IsEntryAssembly(assembly))
+            {
+                context = LoadEntryAssemblyContext();
+            }
+
+            if (context == null)
+            {
+                context = LoadAssemblyContext(assembly);
+            }
+
+            if (context?.IsPortable == true)
+            {
+                var runtimeContext = LoadRuntimeContext();
+                if (runtimeContext != null)
+                {
+                    context = context.Merge(runtimeContext);
+                }
+            }
+            return context;
```

Comment:
Yes, it can be null if no `deps` file was found in any of locations, the same logic as it is today.